internal/clients/vault_secrets.go

```diff
@@ -151,3 +169,16 @@ func DeleteVaultSecretsAppSecret(ctx context.Context, client *Client, loc *share
 
 	return nil
 }
+
+func getAPIBackoffDuration(serviceErr *secret_service.OpenAppSecretDefault) time.Duration {
+	re := regexp.MustCompile(`try again in (\d+) seconds`)
```

Comment:
Do we have access to the response headers? It might be less brittle to fetch the value from `x-ratelimit-reset` header instead (in case the error message changes at some point).

Reply:
I was also trying to access the HTTP headers but couldn't find an easy way to access them on resp/err. Let me know if there a way, and I'll drop the err parsing which is bit brittle.

Reply:
This does seem like a fragile solution. I would definitely recommend changing this to using headers.

Reply:
I think the generated client don't carry the headers as such. I tried typecasting and looking up the header, but couldn't retrieve the value for x-ratelimit-reset. I do see it being printed though.